CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+
+There are no hard rules as to what features will or will not be accepted. It can be a network protocol, integrating with a feature in DPDK, or something new entirely. Ultimately, it depends on what the expected benefit is relative to the expected maintenance burden.
+
+## Pull Requests
+
+Pull requests which fix bugs, add features or improve documentation are welcome and greatly appretiated. Before making a large change, it is usually a good idea to first open an issue describing the change to solicit feedback and guidance. This will increase the likelihood of the PR getting merged.
+
+### Development Notes
+
+Capsule requires extra arguments to many common `cargo` commands you use. We've simplified these commands with a set of [make targets](Makefile) so you don't have to remember them.
+
+To compile the workspace with all features enabled, use
+
+```
+make build
+```
+
+To run the unit tests, including those that are feature-gated, use
+
+```
+make test
+```
+
+We use [rustfmt](https://github.com/rust-lang/rustfmt) to maintain a consistent coding style. To automatically format your source code, use
+
+```
+make fmt
+```
+
+We use [Clippy](https://github.com/rust-lang/rust-clippy) to catch common Rust mistakes and follow best practices like [Rust 2018](https://doc.rust-lang.org/edition-guide/rust-2018/index.html) idioms. To lint the source code, use
+
+```
+make lint
+```
+
+When building documentation normally, the markers that list the features required for various parts of Capsule are missing. To build the documentation correctly, install nightly Rust (`rustup install nightly`) and use
+
+```
+make docs
+```
+
+### Commits
+
+It is a recommended best practice to keep your changes as logically grouped as possible within individual commits. There is no limit to the number of commits any single Pull Request may have. That said, if you have a number of commits that are "checkpoints" and don't represent a single logical change, please squash those together.
+
+The first line of the commit message should
+  * contain a short description of the change no more than 72 characters
+  * start with an imperative verb in the present tense
+  * be entirely in lowercase with the exception of proper nouns and acronyms
+
+When necessary to include a longer commit message, keep the second line blank. You can use markdown syntax for the rest of the commit message.
+
+Note that multiple commits may get squashed when they are merged.
+
+### Opening the Pull Request
+
+Opening a new Pull Request will present you with a [template](.github/pull_request_template.md) that should be filled out. Please try to do your best at filling out the details.
+
+You will probably get feedback or requests for changes to your Pull Request. This is a big part of the submission process so don't be discouraged! This is a necessary part of the process in order to evaluate whether the changes are correct and necessary.
```

Comment:
Look for feedback and/or requests for changes to your Pull Request.

Reply:
seems not necessary? `or` kinda conveys that anyway.